compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -1056,16 +1056,30 @@ main = do
     -- Note: need to parse given args first to decide whether we need to add
     -- args from daml.yaml.
     Command cmd mbProjectOpts _ <- handleParseResult tempParseResult
-    damlYamlArgs <- cliArgsFromDamlYaml mbProjectOpts
-    let args = if cmd `elem` [Build, Compile, Desugar, Ide, DebugIdeSpanInfo, Test, DamlDoc, Repl]
-               then cliArgs ++ damlYamlArgs
-               else cliArgs
+    damlYamlArgs <- if cmd `elem` cmdUseDamlYamlArgs
+      then cliArgsFromDamlYaml mbProjectOpts
+      else pure []
+    let args = cliArgs ++ damlYamlArgs
         (errMsgs, parseResult) = parse args
     Command _ _ io <- handleParseResult parseResult
     forM_ errMsgs $ \msg -> do
         hPutStrLn stderr msg
     withProgName "damlc" io
 
+-- | Commands for which we add the args from daml.yaml build-options.
+cmdUseDamlYamlArgs :: [CommandName]
```

Comment:
it's not my first time fixing a ticket like this, so I thought this would help find the place to change if we get another one.

Reply:
thanks for pointing this out. I looked at the other constructors of `CommandName` and I think `DocTest` and `Lint` probably need to do this too, so I added them. I also changed the type of `cmdUseDamlYamlArgs`, now it's a predicate on `CommandName` so we have to be exhaustive, and I've added comments explaining the `False`s